test/cookies_spec.js

```diff
@@ -1,75 +1,174 @@
+  });
+
+  describe('if response does not contain cookies', function () {
+    it('response.cookies is undefined', function (done) {
+      needle.get('http://nocookies.test', function (error, response) {
+        assert(!response.cookies);
+        done();
+      });
+    });
+  });
+
+  describe('if response contains cookies', function () {
+    it('puts them on resp.cookies', function (done) {
+      needle.get('http://allcookies.test', function (error, response) {
+        response.should.have.property('cookies');
+        done();
+      });
     });
-  })
-
-  after(function(done) {
-    server.close(done)
-  })
-
-  describe('with default options', function() {
-
-    it('no cookie header is set on request', function() {
-
-    })
-
-  })
-
-  describe('if response does not contain cookies', function() {
 
-    it('resp.cookies is undefined', function() {
+    it('parses them as a object', function (done) {
+      needle.get('http://allcookies.test', function (error, response) {
+        response.cookies.should.be.an.instanceOf(Object).and.have.property('wc');
+        response.cookies.should.have.property('bc');
+        response.cookies.should.have.property('fc');
+        done();
+      });
+    });
 
-    })
+    it('must decode it', function (done) {
+      needle.get('http://allcookies.test', function (error, response) {
+        response.cookies.wc.should.be.eql(WEIRD_COOKIE);
+        response.cookies.bc.should.be.eql(BASE64_COOKIE);
+        response.cookies.fc.should.be.eql(FORBIDDEN_COOKIE);
+        done();
+      });
+    });
 
-  })
 
-  describe('if response contains cookies', function() {
+    describe('and response is a redirect', function () {
 
-    it('parses them', function() {
+      describe('and follow_set_cookies is false', function () {
+        it('no cookie header set on redirection request', function () {});
+      });
 
-    })
+      describe('and follow_set_cookies is true', function () {});
 
-    it('puts them on resp.cookies', function() {
+    });
 
-    })
+  });
 
-    describe('and response is a redirect', function() {
+  describe('if resquest contains cookie header', function () {
+    var opts;
 
-      describe('and follow_set_cookies is false', function() {
+    before(function () {
+      opts = {
+        'cookies': {
+          'weird': WEIRD_COOKIE,
+          'base': BASE64_COOKIE,
+          'forbidden': FORBIDDEN_COOKIE
+        }
+      };
+    });
 
-      })
+    it('must be a valid cookie string', function (done) {
+      var COOKIE_PAIR = /^([^=\s]+)\s*=\s*("?)\s*(.*)\s*\2\s*$/;
 
-      describe('and follow_set_cookies is true', function() {
-        
-      })
+      needle.get('http://allcookies.test', opts, function (error, response) {
+        var cookieString = response.req._headers.cookie;
 
-    })
+        cookieString.should.be.type('string');
 
-  })
+        cookieString.split(/\s*;\s*/).forEach(function (pair) {
+          COOKIE_PAIR.test(pair).should.be.exactly(true);
```

Comment:
Shoudn't we test for the value themselves rather than just testing against the regex?